### ISSUE TITLE
Current line number is not highlighted as compared to v0.5

### ISSUE THREAD

**Issue by paniash** [neovim/neovim#15379]
### Neovim Version

`v0.6.0-dev+139-g0f2af87a3`

### Operating system/version:

Linux

### Terminal name/version:

Suckless terminal

### TERM environment variable

`st-256color`

### Installation

Build from repo

### Steps to reproduce

Use a colorscheme not provided by the defaults (since all line numbers are brightly colored in default colorschemes). 

Set `set number relativenumber` in `init.vim`.

A distinction exists between the current line number (colored bright) and the rest of the line numbers (comparatively dimmer). This is the case with v0.5.

However, with the version specified above, all line numbers are colored the same. The main issue being that the current line number is colored the same as the other ones (i.e. dimmer).

### Vim

Same issue seems to exist with vim
version 8.2 with included patches: 1-2891

### Expected behavior

![pic-selected-210815-1500-10](https://user-images.githubusercontent.com/59497618/129475239-93c12129-acc9-4a79-91c0-fc3b4b14095a.png)

### Actual behavior

![pic-selected-210815-1524-41](https://user-images.githubusercontent.com/59497618/129475242-546b1829-1a4f-4146-bbbc-4cdd6e3c1054.png)

TL;DR The highlight group `CursorLineNr` does not work.

**Comment by zeertzjq** [comment]
Use `:set cursorline cursorlineopt=number`.

**Comment by paniash** [comment]
@zeertzjq Thanks! This works. :-)

**Comment by zeertzjq** [comment]
Maybe this is a documentation problem. <https://github.com/vim/vim/pull/8763>

**Comment by paniash** [comment]
I feel, this should be made the default.

**Comment by paniash** [comment]
@zeertzjq For certain filetypes like `netrw`, it is preferable to have the default style of `cursorline` set (i.e. the whole line highlighted). Is there a way to do this? A very hacky way would be:

```vim
  au FileType dirvish set cursorline cursorlineopt=both
```

But this also means that any file I open with netrw gives me the same cursorline for that filetype too.

**Comment by zeertzjq** [comment]
Maybe things can be easier if [8.1.2229](https://github.com/vim/vim/commit/efae76ab1a43d5a628d8c2fa4218ace6ba597f5d) is ported.

**Comment by gpanders** [comment]
@paniash `'both'` (e.g. `'number,line'` is already the default).

**Comment by zeertzjq** [comment]
#15409
